src/mpyl/cli/commands/projects/lint.py

```diff
@@ -124,11 +127,48 @@ def __detail_wrong_substitutions(
         project.name.lower(): project.name for project in all_projects
     }
     for project in wrong_substitutions_per_project:
-        console.print(f"❌ Project {project.name} has wrong namespace substitutions:")
+        console.print(f"  ❌ Project {project.name} has wrong namespace substitutions:")
         for env, url in project.wrong_substitutions:
             unrecognized_project_name = url.split(".{namespace}")[0].split("/")[-1]
             suggestion = all_project_names.get(unrecognized_project_name.lower())
             console.print(
                 f"  {env} references unrecognized project {unrecognized_project_name}"
                 + (f" (did you mean {suggestion}?)" if suggestion else "")
             )
+
+
+def _lint_whitelisting_rules(
+    console: Console,
+    projects: list[Project],
+    config: dict,
+    target: Target,
+):
+    console.print("")
+    console.print("Checking whitelisting rules: ")
+    defined_whitelists: set[str] = set(
+        map(lambda rule: rule["name"], config["whiteLists"]["addresses"])
+    )
+    wrong_whitelists: list[tuple[Project, set]] = []
+    for project in projects:
+        if project.deployment:
+            if traefik := project.deployment.traefik:
+                whitelists: set[str] = set(
+                    itertools.chain.from_iterable(
+                        [
+                            whitelist_property.get_value(target)
+                            for whitelist_property in [
+                                host.whitelists
+                                for host in traefik.hosts
+                                if host.whitelists is not None
+                            ]
+                            if whitelist_property.get_value(target) is not None
+                        ]
+                    )
+                )
+                if diff := whitelists.difference(defined_whitelists):
+                    wrong_whitelists.append((project, diff))
+    if len(wrong_whitelists) == 0:
+        console.print("  ✅ No undefined whitelists found")
+    else:
+        for project, diff in wrong_whitelists:
+            console.log(f"  ❌ Project {project.name} has undefined whitelists: {diff}")
```

Comment:
Print diff with pretty print https://github.com/Vandebron/mpyl/blob/f9cf247dbaa0a987e31de61989b7602fcb4c6c59/src/mpyl/projects/versioning.py#L299

Reply:
Hm diff here is a set, DeepDiff doesn't seem to work for those